backend/services/studentService.js

```diff
@@ -28,8 +29,21 @@ module.exports = {
         }
         return studentDAO.setStudentSkills(studentId, skills)    
     },
-    // getStudentsBySkills: (skills) => {
-    //     let skillIds = skills.map(skill => skill.skill_id);
-    //     return studentDAO.getStudentsBySkills(skillIds);
-    // },
+
+    getStudentsBySkills: async (skills) => {
+        let skillIds = skills.map(skill => skill.skill_id);
+        return await studentDAO.getStudentsBySkills(skillIds);
+    },
+
+    getStudentProfile: async (id) => {
+        let profile = await studentDAO.getStudentById(id);
+        profile = profile[0];
+        let skillIds = await skillsDAO.getStudentSkillsById(profile.id);
```

Comment:
The student service should not directly access DAO of skills. It should request whatever it wants from skillService. The services communicate with each other.
Besides this, we are making 2 queries here to the same table to get a piece of data that could have been very easily taken in only 1 query. 

So we should have something like: 
`profile.skills = await skillsService.getStudentSkills(profile.id)`

and in the getStudentSkills we should call a method in DAO that gets all the student skills by the student id.

and that method should make a join query on the db between "student_has_skills" and "skills" table, and return the skills in which the student_id is equal with the studentID param

Reply:
I changed this to the version you suggested, and also changed the code elsewere I did the same mistake as here